stable_baselines/gail/dataset/record_expert.py

```diff
@@ -101,22 +104,32 @@ def generate_expert_traj(model, save_path, env=None, n_timesteps=0,
             observations.append(obs)
 
         if isinstance(model, BaseRLModel):
-            action, _ = model.predict(obs)
+            action, _ = model.predict(obs, state=state)
```

Comment:
I have to check for this one, but it should be something like:

```python
action, state = model.predict(obs, state=state, mask=done)
```
or at least:
```python
action, state = model.predict(obs, state=state)
```

Reply:
I am at that point quite sorry for the amount of Time you had to sacrifice, for my stupid mistakes.